### DISCUSSION TITLE
Does @Autowired work in my spring-boot code that is placed in processors on Karavan

### DISCUSSION THREAD

**Discussion by MN-Dheeraj** [apache/camel-karavan#1236] [Q&A] [answered]
### Ask a question

I am building my spring boot code on karavan. I have a custom processor that has class A where object of another class B is declared as member and autowired. This injected object b is used to call methods of B.
However, this doesn't happen as it usually happens in spring boot projects. I get null pointer exception when I call methods of B using b(that is autowired).
Can someone please help me on this?

### Variant

Web Application

### Container Management (if applicable)

Docker

### Operating System (if applicable)

Windows

### Version

4.4.0

**Comment by davsclaus** [comment, marked as answer]
No karavan only runs vanilla camel (camel-main) and will only do dependency injection for _camel stuff_ and among your own .java files where you can use annotations (camel, spring or cdi/quarkus). 

You cannot use spring boot or spring boot auto-configuration or anything like that.
